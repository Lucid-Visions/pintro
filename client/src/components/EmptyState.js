import React, { Component } from "react";
import { View, Image, StyleSheet, Text, Dimensions, TouchableOpacity } from "react-native";
import WideButton from "./WideButton";
import sources, { text, actions } from "../assets/emptyStates/emptyStates";

function EmptyState({ state, onPress, modal }) {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{text[state].h1}</Text>
      <Text style={styles.text2}>{text[state].h2}</Text>
      <Image source={sources[state]} style={modal ? styles.imageModal : styles.image} />
      <View style={{ width: "100%", position: "absolute", bottom: 40 }}>
        <TouchableOpacity
          onPress={() => {
            onPress?onPress():actions[state]();
          }}
        >
          <WideButton
            containerStyle={{ backgroundColor: "black", width: "80%" }}
            textStyle={{ color: "white", fontSize: 15 }}
            value={text[state].button}
          />
        </TouchableOpacity>
      </View>
    </View>
  );
}
const { width, height } = Dimensions.get("window");

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    height: height - 170
  },
  image: {
    height: 150,
    width: "80%",
    resizeMode: "contain",
    marginTop: 100,
    marginBottom: 100
  },
  imageModal: {
    height: 140,
    resizeMode: "contain",
    marginTop: 50,
    marginBottom: 100
  },
  text: {
    fontFamily: "poppins-semi-bold",
    fontSize: 25,
    color: "black",
    textAlign: "center"
  },
  text2: {
    textAlign: "center",
    fontSize: 13,
    color: "#808080",
    fontFamily: "poppins-regular",
    marginTop: 30
  }
});

export default EmptyState;
