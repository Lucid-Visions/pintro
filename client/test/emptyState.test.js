import React from 'react';
import renderer from 'react-test-renderer';
import EmptyState from '../src/components/EmptyState';

test('Empty state renders correctly', () => {
    const component = renderer.create(<EmptyState state="internet"/>).toJSON();
    expect(component).toMatchSnapshot();
});
