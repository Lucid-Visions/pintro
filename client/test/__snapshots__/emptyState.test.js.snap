// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty state renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "height": 1164,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "poppins-semi-bold",
        "fontSize": 25,
        "textAlign": "center",
      }
    }
  >
    No internet connection
  </Text>
  <Text
    style={
      Object {
        "color": "#808080",
        "fontFamily": "poppins-regular",
        "fontSize": 13,
        "marginTop": 30,
        "textAlign": "center",
      }
    }
  >
    Please check your connection settings
  </Text>
  <Image
    source={1}
    style={
      Object {
        "height": 150,
        "marginBottom": 100,
        "marginTop": 100,
        "resizeMode": "contain",
        "width": "80%",
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 40,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "black",
              "borderRadius": 25,
              "display": "flex",
              "height": 50,
              "justifyContent": "center",
              "marginLeft": "auto",
              "marginRight": "auto",
              "width": "80%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 10,
                  "marginRight": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontFamily": "poppins-medium",
                    "fontSize": 15,
                  }
                }
              >
                Internet Settings
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
